tools/wordcloud/amd/src/redraw_wordcloud.js

```diff
@@ -78,19 +81,62 @@ const getAnswers = async (id) => {
     mtmstate.setAttribute('data-wclastupdated', mtmstate.dataset.contentchangedat);
 
     // Redraw wordcloud.
-    document.getElementById(id).setAttribute('data-answers', JSON.stringify(response.answerlist));
-    redrawwordcloud(id);
+    const container = document.getElementById(id);
+    if (!container) {
+        return;
+    }
+    container.setAttribute('data-answers', JSON.stringify(response.answerlist));
+    ensureObserver(container);
+    redrawwordcloud(container);
 
     return;
 };
 
 /**
  * Redraw the wordcloud.
- * @param {string} id
+ * @param {HTMLElement} container
  */
-function redrawwordcloud(id) {
-    let mtmtcanvas = document.getElementById(id);
+function redrawwordcloud(container) {
+    let mtmtcanvas = container;
     let answers = JSON.parse(mtmtcanvas.dataset.answers);
 
-    WordCloud(mtmtcanvas, { list: answers, weightFactor: 24, color: '#f98012', fontFamily: 'OpenSans' });
+    WordCloud(mtmtcanvas, {
+        list: answers,
+        weightFactor: 24,
+        color: '#f98012',
+        fontFamily: 'OpenSans',
+        classes: 'filter_mathjaxloader_equation',
+    });
+}
+
+/**
+ * Ensure MathJax is triggered when the wordcloud mutates.
+ *
+ * @param {HTMLElement} container
+ */
+function ensureObserver(container) {
+    if (!container || observerRegistry.has(container.id)) {
+        return;
+    }
+
+    const observer = new MutationObserver((mutations) => {
+        if (!document.body.contains(container)) {
+            observer.disconnect();
+            observerRegistry.delete(container.id);
+            return;
+        }
+        let hasAddedNodes = false;
+        for (const mutation of mutations) {
+            if (mutation.addedNodes.length > 0) {
+                hasAddedNodes = true;
+                break;
+            }
+        }
+        if (hasAddedNodes) {
+            notifyFilterContentUpdated([container]);
+        }
```

Comment:
Would it also work to scan for the newly added span elements and only add the filter_mathjaxloader_equation CSS class to the newly added span and trigger the notifyContentUpdated for the newly added span? 

Besides that: No need to first determine "hasAddedNodes" and do an "if..." later on. Just do the notify right after the `if (mutation.addedNodes.length)` and then call `break`

But this seems to work this way, good job! :)